### PR TITLE
feat: add `resolve` option

### PR DESCRIPTION
**Pull request by underfin**
### Test Plan

---

### CODE REVIEW

**Comment by underfin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #392** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/392?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/392?utm_source=stack-comment).

File: crates/rolldown_binding/index.d.ts [hunk 1]
```diff
@@ -30,10 +26,22 @@ export interface InputItem {
   name?: string
   import: string
 }
+export interface ResolveOptions {
+  alias: Record<string, Array<string>>
```
Comment (underfin) [inline crates/rolldown_binding/index.d.ts]: Here need to conside export to js side `Record<String, String>`. Hope get some ideas.

**Review by milesj — APPROVED** [review]

File: crates/rolldown_resolver/src/resolver_options.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        .unwrap_or(options.alias),
+      alias_fields: value.alias_fields.unwrap_or(options.alias_fields),
+      condition_names: value.condition_names.unwrap_or(options.condition_names),
+      exports_fields: value.exports_fields.unwrap_or(options.exports_fields),
+      extensions: value.extensions.unwrap_or(options.extensions),
+      main_fields: value.main_fields.unwrap_or(options.main_fields),
+      main_files: value.main_files.unwrap_or(options.main_files),
+      modules: value.modules.unwrap_or(options.modules),
+      symlinks: value.symlinks.unwrap_or(options.symlinks),
+      ..options
```
Comment (Boshen) [inline crates/rolldown_resolver/src/resolver_options.rs]: I recommend fill in and override all the defaults here so we don't get surprising behaviours when `oxc_resolver` adds more fields or changes its defaults.

Comment (underfin) [reply crates/rolldown_resolver/src/resolver_options.rs]: Fixed it.

**Pull request merged**